Content/GUI/CloseableSmartUi.cs

```diff
@@ -33,7 +33,11 @@ protected override void DrawSelf(SpriteBatch spriteBatch)
         base.DrawSelf(spriteBatch);
 #if DEBUG
         GUIDebuggingTools.DrawGuiBorder(spriteBatch, GetDimensions(), Color.Purple);
-        GUIDebuggingTools.DrawGuiBorder(spriteBatch, Panel.GetDimensions(), Color.Yellow);
+
+		if (Panel is not null)
+		{
+			GUIDebuggingTools.DrawGuiBorder(spriteBatch, Panel.GetDimensions(), Color.Yellow);
```

Comment:
Should we maybe just pass in the entire panel instead of just the dimensions and check this there?

Reply:
What do you mean?

Reply:
```c
GUIDebuggingTools.DrawGuiBorder(spriteBatch, Panel, Color.Yellow);
```
```c
GUIDebuggingTools {
   DrawGuiBorder(....) { 
       if (Panel is not null) { 
          ///do the things
       } 
  }
}
```

Reply:
Oh, yeah. That's definitely easier - speaking of, what if we included the #if preprocessor in DrawGuiBorder instead of around every call?

Reply:
While that would be less code it wouldn't be indicative to the developer that it wouldn't happen if they aren't in debug mode.

Reply:
That's fair then.